### DISCUSSION TITLE
Dependency Graphs in Readme's

### DISCUSSION THREAD

**Discussion by JacksonUtsch** [pointfreeco/swift-composable-architecture#1004] [General]
GitHub recently introduced rendering support for the [Mermaid](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/) language in _markdown_ files. This allows diagrams to be rendered.

The flowchart does well to represent unidirectional flow. I've used it in an application to show how the `Package.swift` is structured. Ultimately this could be made into a GitHub Action and automated using [SwiftPM](https://github.com/apple/swift-package-manager) The purpose is for developers to visualize the dependency graph in their projects.

Here's an example:

## Dependency Graph

```mermaid
  graph TD;
      TCA--> Shared;
      
      Shared-->FeatureA;
      ClientA--> FeatureA;
      RemoteDependencyA--> FeatureA;
      RemoteDependencyB--> FeatureA;
      
      FeatureA--> FeatureATests;
      
      RemoteDependencyA--> FeatureB;
      Shared--> FeatureB;
      
      FeatureB--> FeatureBTests;
      
      FeatureA-->App;
      FeatureB-->App;
      
      App-->AppTests;
```

Would anyone be inclined to use this?

**Comment by JackYoustra** [comment]
This looks so cool! Did you end up writing it up?

**Reply by JacksonUtsch** [reply]
Have only opted for manual usage, haven't taken the time to figure out an action for it.
